### PR TITLE
XmlValidator should properly fail XML files without schema

### PR DESCRIPTION
**Pull request by Happy-Neko**
Unit test for #498.

XmlValidator should properly fail XML files without schema with `ValidationException`. And it does inside `qulice-xml` module that uses `jcabi-xml` version **0.15**. `qulice-maven-plugin` on the other hand uses  `jcabi-xml` version **0.17** and fails the same test with vague `NullPoinerException`.

Easily reproducible by updating `jcabi-xml` version inside `qulice-xml/pom.xml` to **0.17**.

Probably caused by https://github.com/jcabi/jcabi-xml/issues/94.

### CODE REVIEW

**Comment by Happy-Neko** [comment]
@davvd Please proceed with code review.

**Comment by davvd** [comment]
@happy-neko Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@pinaf review this one, please

File: qulice-xml/src/test/java/com/qulice/xml/XmlValidatorTest.java [hunk 1]
```diff
@@ -72,6 +73,22 @@ public void passesValidationOnCorrectFile() throws Exception {
...
+    @Test(expected = ValidationException.class)
+    public void failValidationWithoutSchema() throws Exception {
+        final Environment env = new Environment.Mock()
+            .withFile(
+                "src/main/resources/noschema.xml",
+                // @checkstyle LineLength (1 line)
+                "<Configuration><Appenders><Console name=\"CONSOLE\" target=\"SYSTEM_OUT\"><PatternLayout pattern=\"[%p] %t %c: %m%n\"/></Console></Appenders></Configuration>"
+            );
+        final Validator validator = new XmlValidator(true);
+        validator.validate(env);
```
Comment (pinaf) [inline qulice-xml/src/test/java/com/qulice/xml/XmlValidatorTest.java]: @Happy-Neko how about
```java
new XmlValidator(true).validate(
    new Environment.Mock().withFile(
        ...
    )
);
```

**Comment by pinaf** [comment]
@Happy-Neko nice. 1 comment.

**Comment by krzyk** [comment]
@Happy-Neko so your changes don't fix the problem right? Wouldn't it be better to have invoker test in qulice-maven-plugin, and downgrade jcabi xml there?

**Comment by Happy-Neko** [comment]
@pinaf thanks. Changes committed.

**Comment by Happy-Neko** [comment]
> Wouldn't it be better to have invoker test in qulice-maven-plugin, and downgrade jcabi xml there? 

@krzyk `jcabi-xml` version **0.17** is defined in `jcabi-parent`. We can override it to **0.15** for maven plugin and this bug will be fixed. But I don't know if it causes other regressions. 
I think the better approach is to fix `jcabi-xml` and set **0.18** in `jcabi-parent` to have one dependecy version across the project.

**Comment by krzyk** [comment]
@Happy-Neko I see your point, but even if it will be fixed (which can take months, unless someone will volunteer) we would then need to create another test, but using invoker. So this test is essential, without it we don't know if we fixed, or will fix this issue.

Moreover we didn't help the reporter anyhow, the bug is still there without any slight improvement, so he won't close the issue either way. Because of that I would still try downgrading jcabi-xml.

**Comment by pinaf** [comment]
@krzyk good point - sorry for not picking that up.

**Comment by Happy-Neko** [comment]
@krzyk Bug in 'jcabi-xml` **0.15** prevents from downgrading, Hibernate Validator complains:

    HV000151: A method overriding another method must not alter the parameter constraint configuration, but method public java.util.List com.jcabi.xml.XMLDocument.xpath(java.lang.String) changes the configuration of public abstract java.util.List com.jcabi.xml.XML.xpath(java.lang.String).

Method signatures are fixed in **0.17**.

May be we should move this test to `qulice-maven-plugin`, mark it `@igonred` and postpone task until jcabi-xml bug if fixed?

**Comment by krzyk** [comment]
@Happy-Neko this error is a result of jcabi-aspects bug https://github.com/jcabi/jcabi-aspects/issues/194 so downgrading it should fix it.

**Comment by krzyk** [comment]
@Happy-Neko issue in jcabi-xml has been resolved, new release 0.17.1 has it fixed

**Comment by Happy-Neko** [comment]
@krzyk version updated. Bug #498 is fixed.

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by krzyk** [comment]
@Happy-Neko thanks

**Comment by davvd** [comment]
@pinaf thanks, paid, **18 mins** to your account, payment ID is `AP-9C245162V56633123`, task took 144 hours and 50 mins to complete; review comments (c=3) added as a bonus; added +18 to your rating, now it is equal to [+5208](http://www.netbout.com/b/35128?open=rating)

**Comment by davvd** [comment]
@rultor deploy pls

**Pull request merged**